In [1]:
import pandas as pd
from sklearn.feature_extraction.text import TfidfVectorizer

# 1. Load dataset "data.csv" menjadi dataframe
data = pd.read_csv("data.csv")

In [2]:
data.head()

Unnamed: 0                  userName  \
0           0               Gadri Bandu   
1           1  070_Putri Azzahra Nurdin   
2           2       Novrealita Setiyana   
3           3                auah gelap   
4           4               Adi Nugroho   

                                             content  score  
0  Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1  Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2  Sejauh ini aku main aku suka banget sama model...      5  
3  Semuanya sudah sempurna tapi Sangat disayangka...      3  
4  Grafik bagus,story bagus,terutama archon quest...      5

In [3]:
# 2. Ambil 5 data dan ubah kolom "content" menjadi vektor TF-IDF
sample_data = data.head(5)  # Ambil 5 data pertama
content = sample_data["content"]

In [4]:
# Inisialisasi TF-IDF Vectorizer
tfidf_vectorizer = TfidfVectorizer()

# Transformasi teks menjadi vektor TF-IDF
tfidf_matrix = tfidf_vectorizer.fit_transform(content)

In [5]:
# Tampilkan hasilnya
print("Hasil TF-IDF Vector untuk 5 data pertama:")
print(tfidf_matrix)

Hasil TF-IDF Vector untuk 5 data pertama:
  (0, 143)	0.1228829916017791
  (0, 112)	0.15231030072560398
  (0, 22)	0.15231030072560398
  (0, 9)	0.15231030072560398
  (0, 41)	0.10200398350433568
  (0, 73)	0.10200398350433568
  (0, 105)	0.1228829916017791
  (0, 78)	0.1228829916017791
  (0, 156)	0.15231030072560398
  (0, 83)	0.15231030072560398
  (0, 55)	0.15231030072560398
  (0, 27)	0.14515334876102162
  (0, 138)	0.15231030072560398
  (0, 152)	0.15231030072560398
  (0, 124)	0.15231030072560398
  (0, 88)	0.30462060145120795
  (0, 12)	0.15231030072560398
  (0, 16)	0.15231030072560398
  (0, 35)	0.1228829916017791
  (0, 92)	0.15231030072560398
  (0, 15)	0.10200398350433568
  (0, 111)	0.15231030072560398
  (0, 148)	0.15231030072560398
  (0, 188)	0.2457659832035582
  (0, 128)	0.07257667438051081
  :	:
  (4, 147)	0.11990957128388831
  (4, 8)	0.11990957128388831
  (4, 185)	0.11990957128388831
  (4, 5)	0.09674228709978505
  (4, 31)	0.09674228709978505
  (4, 51)	0.09674228709978505
  (4, 194)	0.0803

In [6]:
# 3. Tampilkan nama fiturnya dengan get_feature_names_out dan Compute TF-IDF
feature_names = tfidf_vectorizer.get_feature_names_out()
print("\nNama fitur:")
print(feature_names)


Nama fitur:
['ada' 'agak' 'akan' 'akhirnya' 'aku' 'android' 'apalagi' 'app' 'archon'
 'atasi' 'atau' 'background' 'bagaimana' 'bagus' 'banget' 'banyak'
 'baterei' 'beda²' 'bener' 'bener2' 'berat' 'berharap' 'berhenti' 'besar'
 'bgtt' 'bikin' 'bintang' 'bisa' 'buat' 'bukan' 'character' 'controller'
 'cukup' 'cuma' 'dalam' 'dan' 'dapat' 'dapet' 'dari' 'dengan' 'device'
 'di' 'diatasi' 'didapatkan' 'diperbaiki' 'disayangkan' 'ditunggu' 'dll'
 'dr' 'dua' 'dulu' 'dunia' 'elemen' 'event' 'free' 'full' 'ga' 'gak'
 'gambar' 'game' 'gameplay' 'gamers' 'gampang' 'gemenya' 'genshin'
 'gimana' 'grafik' 'gt' 'habis' 'hadiah' 'hehe' 'ijo' 'in' 'ini' 'itu'
 'jadi' 'jaringan' 'jenis' 'jika' 'juga' 'kalo' 'karakter' 'karakternya'
 'karena' 'karna' 'kasih' 'ke' 'ke3' 'kedepan' 'ketika' 'kita' 'kuning'
 'kuota' 'kurang' 'kurangnya' 'lain' 'lebih' 'level' 'loh' 'luas' 'maen'
 'main' 'makin' 'malah' 'mana' 'masalah' 'masuk' 'mati' 'mau' 'memang'
 'menarik' 'menghabiskan' 'menjadi' 'menyelesaikan' 'mic' 'm

In [7]:
# Compute Term Frequency (TF) dan Inverted Document Frequency (IDF)
for i, data_idx in enumerate(sample_data.index):
    print(f"\nData ke-{i + 1}:")
    data_tf = tfidf_matrix[i].toarray()  # Mendapatkan vektor TF-IDF untuk data tertentu
    for j, feature_name in enumerate(feature_names):
        print(f"{feature_name}: {data_tf[0][j]:.4f}")


Data ke-1:
ada: 0.0000
agak: 0.0000
akan: 0.0000
akhirnya: 0.0000
aku: 0.0000
android: 0.0000
apalagi: 0.0000
app: 0.0000
archon: 0.0000
atasi: 0.1523
atau: 0.0000
background: 0.0000
bagaimana: 0.1523
bagus: 0.1020
banget: 0.0000
banyak: 0.1020
baterei: 0.1523
beda²: 0.0000
bener: 0.0000
bener2: 0.0000
berat: 0.1229
berharap: 0.0000
berhenti: 0.1523
besar: 0.0000
bgtt: 0.0000
bikin: 0.0000
bintang: 0.1523
bisa: 0.1452
buat: 0.0000
bukan: 0.0000
character: 0.0000
controller: 0.0000
cukup: 0.0000
cuma: 0.0000
dalam: 0.0000
dan: 0.1229
dapat: 0.0000
dapet: 0.0000
dari: 0.0000
dengan: 0.0000
device: 0.0000
di: 0.1020
diatasi: 0.0000
didapatkan: 0.0000
diperbaiki: 0.0000
disayangkan: 0.0000
ditunggu: 0.0000
dll: 0.0000
dr: 0.0000
dua: 0.0000
dulu: 0.1523
dunia: 0.0000
elemen: 0.0000
event: 0.0000
free: 0.0000
full: 0.1523
ga: 0.0000
gak: 0.0000
gambar: 0.0000
game: 0.2040
gameplay: 0.0000
gamers: 0.0000
gampang: 0.0000
gemenya: 0.1523
genshin: 0.0000
gimana: 0.0000
grafik: 0.0000
gt: 0.000

In [8]:
# Buat dataframe dari matriks TF-IDF dengan nama kolom sesuai dengan fitur kata
tfidf_df = pd.DataFrame(data=tfidf_matrix.toarray(), columns=feature_names)

# Tampilkan matriks kata
print(tfidf_df)

        ada      agak      akan  akhirnya       aku   android  apalagi  \
0  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.00000   
1  0.000000  0.000000  0.000000  0.000000  0.112449  0.000000  0.00000   
2  0.071286  0.265072  0.000000  0.088357  0.213858  0.000000  0.00000   
3  0.219425  0.000000  0.135986  0.000000  0.000000  0.109713  0.00000   
4  0.000000  0.000000  0.000000  0.000000  0.000000  0.096742  0.11991   

        app   archon    atasi  ...   tolong       tuh      udh     untuk  \
0  0.000000  0.00000  0.15231  ...  0.00000  0.000000  0.00000  0.000000   
1  0.000000  0.00000  0.00000  ...  0.00000  0.000000  0.00000  0.000000   
2  0.000000  0.00000  0.00000  ...  0.00000  0.088357  0.00000  0.059174   
3  0.135986  0.00000  0.00000  ...  0.00000  0.000000  0.00000  0.091071   
4  0.000000  0.11991  0.00000  ...  0.11991  0.000000  0.11991  0.080305   

         up   update      wkwk     world     yang        yg  
0  0.000000  0.15231  0.000000  0.00

__REPRESENTASI__

Indeks 0:

- Kata "aku" memiliki bobot TF-IDF sebesar 0.0, yang berarti kata ini tidak muncul dalam dokumen ini.

- Kata "akan" juga memiliki bobot TF-IDF sebesar 0.0.

- Kata "atasi" memiliki bobot TF-IDF sebesar 0.15231, menunjukkan bahwa kata ini penting dalam dokumen ini.

- Kata-kata lainnya juga tidak muncul dalam dokumen ini, diberi bobot TF-IDF 0.0.


Indeks 1:

- Kata "aku" memiliki bobot TF-IDF sebesar 0.112449, menunjukkan bahwa kata ini muncul dalam dokumen ini dan memiliki tingkat penting yang sedang.

- Kata "world" memiliki bobot TF-IDF sebesar 0.139378, yang menunjukkan pentingnya kata ini dalam dokumen ini.

- Kata "yg" memiliki bobot TF-IDF tertinggi sebesar 0.224898, menunjukkan tingkat penting yang tinggi.

- Kata-kata lainnya juga tidak muncul dalam dokumen ini, diberi bobot TF-IDF 0.0.

Indeks 2:

- Kata "ada" memiliki bobot TF-IDF sebesar 0.071286, yang menunjukkan bahwa kata ini muncul dalam dokumen ini dan memiliki tingkat penting yang sedang.

- Kata "agak" memiliki bobot TF-IDF sebesar 0.265072, yang menunjukkan tingkat penting yang tinggi dalam dokumen ini.

- Kata "akhirnya" memiliki bobot TF-IDF sebesar 0.088357, menunjukkan bahwa kata ini juga cukup penting.

- Kata-kata lainnya yang tidak muncul dalam dokumen ini diberi bobot TF-IDF 0.0.

Indeks 3:

- Kata "ada" memiliki bobot TF-IDF sebesar 0.219425, menunjukkan tingkat penting yang tinggi dalam dokumen ini.

- Kata "akan" memiliki bobot TF-IDF sebesar 0.135986, menunjukkan tingkat penting yang sedang dalam dokumen ini.

- Kata "app" memiliki bobot TF-IDF sebesar 0.135986, menunjukkan tingkat penting yang sedang dalam dokumen ini.

- Kata "untuk" memiliki bobot TF-IDF sebesar 0.091071, menunjukkan bahwa kata ini juga cukup penting.

- Kata-kata lainnya yang tidak muncul dalam dokumen ini diberi bobot TF-IDF 0.0.

Indeks 4

- Kata "android" memiliki bobot TF-IDF sebesar 0.096742, menunjukkan tingkat penting yang sedang dalam dokumen ini.

- Kata "archon" memiliki bobot TF-IDF sebesar 0.11991, menunjukkan bahwa kata ini cukup penting.

- Kata "tolong" dan "udh" juga memiliki bobot TF-IDF sebesar 0.11991, menunjukkan pentingnya kata-kata ini dalam dokumen ini.

- Kata-kata lainnya yang tidak muncul dalam dokumen ini diberi bobot TF-IDF 0.0.